.github/workflows/cli.yml

```diff
@@ -0,0 +1,59 @@
+name: cli
+
+on:
+  push:
+    branches:
+      - main
+  pull_request: {}
+
+concurrency:
+  group: cli-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Build with Node ${{ matrix.node }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [ '12', '14', '16' ]
+    steps:
```

Comment:
@lascenify let me know if you have any questions regarding this. It's a sequence of steps that gets executed sequentially.